server/ast/expr.go

```diff
@@ -0,0 +1,517 @@
+	case *tree.DOid:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DOidWrapper:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DString:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DTime:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DTimeTZ:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DTimestamp:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DTimestampTZ:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DTuple:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DUuid:
+		return nil, fmt.Errorf("the statement is not yet supported")
+	case *tree.DefaultVal:
+		return nil, fmt.Errorf("default values are not yet supported")
+	case *tree.FuncExpr:
+		return nodeFuncExpr(node)
+	case *tree.IfErrExpr:
+		return nil, fmt.Errorf("IFERROR is not yet supported")
+	case *tree.IfExpr:
+		//TODO: probably should be the IF function
+		return nil, fmt.Errorf("IF is not yet supported")
+	case *tree.IndexedVar:
+		//TODO: figure out if I can delete this
+		return nil, fmt.Errorf("this should probably be deleted (internal error, IndexedVar)")
+	case *tree.IndirectionExpr:
+		return nil, fmt.Errorf("subscripts are not yet supported")
+	case *tree.IsNotNullExpr:
+		expr, err := nodeExpr(node.Expr)
+		if err != nil {
+			return nil, err
+		}
+		return &vitess.IsExpr{
+			Operator: vitess.IsNotNullStr,
+			Expr:     expr,
+		}, nil
+	case *tree.IsNullExpr:
+		expr, err := nodeExpr(node.Expr)
+		if err != nil {
+			return nil, err
+		}
+		return &vitess.IsExpr{
+			Operator: vitess.IsNullStr,
+			Expr:     expr,
+		}, nil
+	case *tree.IsOfTypeExpr:
+		return nil, fmt.Errorf("IS OF is not yet supported")
+	case *tree.NotExpr:
+		expr, err := nodeExpr(node.Expr)
+		if err != nil {
+			return nil, err
+		}
+		return &vitess.NotExpr{
+			Expr: expr,
+		}, nil
+	case *tree.NullIfExpr:
+		//TODO: probably should be the IF function: IF(Expr1 == Expr2, NULL, Expr1)
+		return nil, fmt.Errorf("IFERROR is not yet supported")
+	case *tree.NumVal:
+		switch node.Kind() {
+		case constant.Int:
+			return vitess.NewIntVal([]byte(node.FormattedString())), nil
+		case constant.Float:
+			return vitess.NewFloatVal([]byte(node.FormattedString())), nil
+		default:
+			return nil, fmt.Errorf("unknown number format")
+		}
+	case *tree.OrExpr:
+		left, err := nodeExpr(node.Left)
+		if err != nil {
+			return nil, err
+		}
+		right, err := nodeExpr(node.Right)
+		if err != nil {
+			return nil, err
+		}
+		return &vitess.OrExpr{
+			Left:  left,
+			Right: right,
+		}, nil
+	case *tree.ParenExpr:
+		expr, err := nodeExpr(node.Expr)
+		if err != nil {
+			return nil, err
+		}
+		return &vitess.ParenExpr{
+			Expr: expr,
+		}, nil
+	case *tree.PartitionMaxVal:
+		return nil, fmt.Errorf("MAX is not yet supported")
```

Comment:
This message and the next are both wrong, should be `MAXVALUE`, `MINVALUE`

Reply:
I looked this up at the time and got confused, the documentation was mentioning `MAX` and `MIN`, but they were under a slightly different context.